### PR TITLE
Move `mbedtls_dev` to framework submodule

### PR DESCRIPTION
**Pull request by davidhorstmann-arm**
Move the `mbedtls_dev` python module into the `mbedtls-framework` repository. This PR pairs with Mbed-TLS/mbedtls-framework#15.

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** not required - non-user-facing change
- [x] **3.6 backport** #9131 
- [x] **2.28 backport** not required - no framework submodule for 2.28
- [x] **tests** not required - internal script motion only - current tests should suffice

### CODE REVIEW

File: tests/scripts/scripts_path.py [hunk 1]
```diff
@@ -15,3 +15,6 @@
 sys.path.append(os.path.join(os.path.dirname(__file__),
                              os.path.pardir, os.path.pardir,
                              'scripts'))
+sys.path.append(os.path.join(os.path.dirname(__file__),
```
Comment (ronald-cron-arm) [inline tests/scripts/scripts_path.py:18]: Isn't it the case that we should rather remove this file and change files that import scripts_path to rather import framework_path?

File: scripts/framework_path.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+"""Add our Python library directory to the module search path.
```
Comment (ronald-cron-arm) [inline scripts/framework_path.py:1]: Rather framework_script_path.py for the name of the file?

**Review by ronald-cron-arm — CHANGES_REQUESTED** [review]
This looks quite good to me. I can see that the "Move some files to framework repository" commits in mbedtls and mbedtls-framework are both based on the same commit: 61734ec61d. The one in mbedtls removes the `mbedtls_dev` files while the one in mbedtls-framework removes all files but the `mbedtls_dev` ones that are renamed (path change only) (checked with git show 75541028a9639a8c8cb7805fdec114dc50844931 --name-status in the framework submodule).

Otherwise, have you checked that you can play the same move game subsequently on another set of `mbedtls` files, `tests/data_files/*` for example without merge trouble in `framework`?

**Comment by davidhorstmann-arm** [comment]
> Otherwise, have you checked that you can play the same move game subsequently on another set of `mbedtls` files, `tests/data_files/*` for example without merge trouble in `framework`?

I have now added to the script so that it can do this. The result is in Mbed-TLS/mbedtls-framework#18, which is based on the corresponding PR in the framework submodule.

File: tests/scripts/scripts_path.py (hunk 1 above)
Comment (davidhorstmann-arm) [reply tests/scripts/scripts_path.py:18]: In practice I believe we need both files for 2 reasons:

1. Each file needs to be in the same directory as the scripts that import it, so we need one for `scripts/` and one for `tests/scripts`.
2. Our test scripts still need to import modules from `scripts` that aren't in `mbedtls_dev`, so have not been moved to the framework. For example, `tests/scripts/depends.py` imports `scripts/config.py`.

Comment (ronald-cron-arm) [reply tests/scripts/scripts_path.py:18]: Yes that makes sense. Thanks for writing this down.

**Review by ronald-cron-arm — APPROVED** [review]
This looks good to me, thanks. There is a conflict with development now thus you will need to rebase though.

**Review by ronald-cron-arm — APPROVED** [review]
I've checked the rebase. LGTM, thanks.

**Review by bensze01 — DISMISSED** [review]
LGTM. I've also checked the moved scripts, and they are up to date with the tip of development.

**Comment by ronald-cron-arm** [comment]
@davidhorstmann-arm @bensze01 I've just updated the framework to the merge of #15.

**Review by bensze01 — APPROVED** [review]
LGTM.

**Pull request merged**